src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -198,7 +204,12 @@ public override bool TryLoadKnownConfig([NotNullWhen(true)] out RuntimeConfig? c
     public void HotReloadConfig(string defaultDataSourceName, ILogger? logger = null)
     {
         logger?.LogInformation(message: "Starting hot-reload process for config: {ConfigFilePath}", ConfigFilePath);
-        TryLoadConfig(ConfigFilePath, out _, replaceEnvVar: true, defaultDataSourceName: defaultDataSourceName);
+        if (!TryLoadConfig(ConfigFilePath, out _, replaceEnvVar: true, defaultDataSourceName: defaultDataSourceName))
+        {
+            logger?.LogWarning(message: "Unable to hot-reload configuration file due to failure to parse the file");
+        }
+
+        SendEventNotification("Sending Hot-Reload event notification.");
```

Comment:
I made it intentionally, since I thought it would be a good idea to also see if we get any exception from the validation and the parsing. Although, thinking about it once more, I will change it so the TryLoadConfig stops if it fails.